tests/e2e/test_examples.py

```diff
@@ -13,12 +13,21 @@ def test_all_examples_are_runnable(mocker, with_crd, exampledir):
     # If the example has its own opinion on the timing, try to respect it.
     # See e.g. /examples/99-all-at-once/example.py.
     example_py = exampledir / 'example.py'
-    m = re.search(r'^E2E_CREATE_TIME\s*=\s*(\d+)$', example_py.read_text(), re.M)
+    m = re.search(r'^E2E_CREATE_TIME\s*=\s*(.+)$', example_py.read_text(), re.M)
     e2e_create_time = eval(m.group(1)) if m else None
-    m = re.search(r'^E2E_DELETE_TIME\s*=\s*(\d+)$', example_py.read_text(), re.M)
+    m = re.search(r'^E2E_DELETE_TIME\s*=\s*(.+)$', example_py.read_text(), re.M)
     e2e_delete_time = eval(m.group(1)) if m else None
-    m = re.search(r'^E2E_TRACEBACKS\s*=\s*(\w+)$', example_py.read_text(), re.M)
+    m = re.search(r'^E2E_TRACEBACKS\s*=\s*(.+)$', example_py.read_text(), re.M)
     e2e_tracebacks = eval(m.group(1)) if m else None
+    m = re.search(r'^E2E_SUCCESS_COUNTS\s*=\s*(.+)$', example_py.read_text(), re.M)
+    e2e_success_counts = eval(m.group(1)) if m else None
+    m = re.search(r'^E2E_FAILURE_COUNTS\s*=\s*(.+)$', example_py.read_text(), re.M)
+    e2e_failure_counts = eval(m.group(1)) if m else None
+    m = re.search(r'@kopf.on.create\(', example_py.read_text(), re.M)
+    e2e_test_creation = bool(m)
+    m = re.search(r'@kopf.on.(create|update|delete)\(', example_py.read_text(), re.M)
+    e2e_test_highlevel = bool(m)
+
     # check whether there are mandatory deletion handlers or not
     m = re.search(r'@kopf\.on\.delete\((\s|.*)?(optional=(\w+))?\)', example_py.read_text(), re.M)
```

Comment:
This `(\s|.*)?` seems unnecessarily complicated, as it is equivalent to `.*` with the `re.S` flag, creates an unused capturing group, and doesn't match multiple newlines, comments, etc.

Reply:
I do not like this approach with the source code parsing with regular expressions at all. Especially when new criteria are added, as in this case. This "examples as e2e tests" approach should be revised to something else, but I don't know to what (yet). Any ideas are welcome. The goal is to test that the examples are still relevant and functional, while keeping in mind that they all are different.